crates/burn-jit/src/compute/kernel.rs

```diff
@@ -61,52 +109,85 @@ impl WorkGroup {
     }
 }
 
-/// Wraps a [dynamic kernel source](DynamicKernelSource) into a [kernel](Kernel) with launch
-/// information such as [workgroup](WorkGroup).
-#[derive(new)]
-pub struct DynamicKernel<K> {
-    kernel: K,
+/// Wraps a [dynamic jit kernel](DynamicJitKernel) into a [Jit kernel](JitKernel) with launch
+/// information
+pub struct DynamicKernel<K, C> {
+    shader: ComputeShader,
+    id: String,
     workgroup: WorkGroup,
+    _kernel: PhantomData<K>,
+    _compiler: PhantomData<C>,
 }
```

Comment:
We would not need that if we have `GpuJitKernel`, but in this case we don't want any generic, only function generic instead of struct.